static/app/utils/profiling/profile/chromeTraceProfile.tsx

```diff
@@ -285,3 +289,128 @@ export function parseChromeTraceArrayFormat(
     profiles,
   };
 }
+
+function isProfileEvent(event: ChromeTrace.Event): event is ChromeTrace.ProfileEvent {
+  return event.ph === 'P' && event.name === 'Profile';
+}
+
+function isProfileChunk(
+  event: ChromeTrace.Event
+): event is ChromeTrace.ProfileChunkEvent {
+  return event.ph === 'P' && event.name === 'ProfileChunk';
+}
+
+function isThreadmetaData(
+  event: ChromeTrace.Event
+): event is ChromeTrace.ThreadMetadataEvent {
+  event.name === 'Thread';
+
+  return event.ph === 'M' && event.name === 'Thread';
+}
+
+type Required<T> = {
+  [P in keyof T]-?: T[P];
+};
+
+// This mostly follows what speedscope does for the Chrome Trace format, but we do minor adjustments (not sure if they are correct atm),
+// but the protocol format seems out of date and is not well documented, so this is a best effort.
+function collectEventsByProfile(input: ChromeTrace.ArrayFormat): {
+  profiles: Map<string, Required<ChromeTrace.CpuProfile>>;
+  threadNames: Map<string, string>;
+} {
+  const sorted = input.sort(chronologicalSort);
+
+  const threadNames = new Map<string, string>();
+  const profileIdToProcessAndThreadIds = new Map<string, [number, number]>();
+  const profiles = new Map<string, Required<ChromeTrace.CpuProfile>>();
+
+  for (let i = 0; i < sorted.length; i++) {
+    const event = sorted[i];
+
+    if (isThreadmetaData(event)) {
+      threadNames.set(`${event.pid}:${event.tid}`, event.args.name);
+      continue;
+    }
+
+    // A profile entry will happen before we see any ProfileChunks, so the order here matters
```

Comment:
Related to the comment about sorting, can the profile event and the profile chunk event have the same timestamp? If so, we need to be careful here since the sort above doesn't give us any guarantee that the profile event will come before the profile chunk event

Reply:
I dont think so, but we can throw in this case and see if it happens :)